### ISSUE TITLE
no-unused-vars thrown even when used in a template string

### ISSUE THREAD

**Issue by Nithanaroy** [eslint/eslint#8256]
**Tell us about your environment**

* **ESLint Version:**^3.15.0
* **Node Version:**7.2.0
* **npm Version:**3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**

```




```

**What did you do? Please include the actual source code causing the issue.**

```js

function formatXHRResponse ( xhrResponse ) {
    return `$(xhrResponse.status) $(xhrResponse.statusText): $(xhrResponse.responseText)`;
  }



```

**What did you expect to happen?**
No error should be thrown by eslint as I used xhrResponse variable
**What actually happened? Please include the actual, raw output from ESLint.**
`no-unused-vars` errors thrown
